src/core/createEventManager.js

```diff
@@ -83,12 +94,45 @@ export default ({ createEvent, optIn, lifecycle, network, config, logger }) => {
           return lifecycle.onBeforeDataCollection({ payload });
         })
         .then(() => {
-          return network.sendRequest(payload, {
+          const endpointDomain = payload.shouldUseIdThirdPartyDomain
+            ? ID_THIRD_PARTY_DOMAIN
+            : edgeDomain;
+          cookieTransfer.cookiesToPayload(payload, endpointDomain);
+          return network.sendRequest(payload, endpointDomain, {
             expectsResponse: payload.expectsResponse,
-            documentUnloading: event.isDocumentUnloading,
-            useIdThirdPartyDomain: payload.shouldUseIdThirdPartyDomain
+            documentUnloading: event.isDocumentUnloading
           });
         })
+        .catch(error => {
+          // The error that we caught is more important than
+          // any error that may have occurred in lifecycle.onResponseError().
+          // For that reason, we make sure the caught error is the one that
+          // bubbles up. We also wait until lifecycle.onRequestFailure is
+          // complete before returning, so that any error that may occur
+          // in lifecycle.onRequestFailure is properly suppressed if the
+          // user has errorsEnabled: false in the configuration.
+          // We could use finally() here, but just to be safe, we don't,
+          // because finally() is only recently supported natively and may
+          // not exist in customer-provided promise polyfills.
+          const throwError = () => {
+            throw error;
+          };
+          return lifecycle.onRequestFailure().then(throwError, throwError);
+        })
+        .then(parsedBody => {
+          const response = createResponse(parsedBody);
+          cookieTransfer.responseToCookies(response);
+
+          // TODO Document that onResponse will be called when Konductor
+          // sends a well-formed response even if that response contains
+          // error objects. This is because even when there are error objects
+          // there can be "handle" payloads to act upon. Also document
+          // that onRequestFailure will be called when the network request
+          // itself failed (e.g., no internet connection), when JAG throws an
+          // error (the request never made it to Konductor), or when
+          // Konductor returns a malformed response.
+          return lifecycle.onResponse({ response }).then(() => response);
```

Comment:
There's a bit of fuzziness around the role of `createNetwork.js` vs `createEventManager.js`, like which one should be responsible for calling `lifecycle.onResponse` (is it too high-level for something as low-level as `createNetwork`?) and which one should be responsible for transferring data from `meta.state` to cookies and vice-versa. There also some unknowns around whether `createEventManager.js` will be the only module to interact with the network object and how things might look if that's not the case. I think when we implement the interaction with an opt-in endpoint we'll discover more.

Reply:
What is your reasoning for putting the cookieTransfer call here instead of in the identity component? I was quite confused by the identity component's hasEcid() method until I saw that the cookie was being transferred here.

Reply:
It can deal with anything that needs to be persisted for the user and not just identity. For this reason, I felt it was lower-level than identity.

Reply:
Yes, but now the identity component isn't worried so much about identity (it doesn't even know what the ecid is!), but about making sure there is a cookie.  What if we had a cookie component that took care of this?

I feel like the "cookie management" and "holding up requests to get first cookie" should be in a component  together because it can do its thing by just handling the lifecycle methods.  Adding functionality to core should be limited to things that need to be shared by multiple components or things that need more than just the lifecycle methods.

Reply:
There are several ways to slice this and some depend on whether Identity can remain blissfully unaware of, specifically, the ECID cookie. Let's talk about this in tomorrow's engineering check-in.